docs/software-platforms/docker/docker-images.md

```diff
@@ -0,0 +1,120 @@
+
+Note that creating docker images is not required for contributing to OpenScPCA, but it can be useful for testing and sharing your analysis module with others.
+Each analysis module will ultimately be updated to include a Docker image before producing final results, but this does not need to be done by the module author.
+Data Lab staff will create a Docker image as part of the process of adding the module to the `OpenScPCA-nf` workflow if it has not already been done.
+
+## Prerequisites
+
+Before making a Docker image, you will want to have [defined the software environment](../../contributing-to-analyses/determining-requirements/determining-software-requirements.md) that you will be using for your analysis module.
+Having the requirements prepared will make creating a fully reproducible Docker image much more straightforward.
+
+For the easiest setup, you should have one of the two following kind of environment definitions:
+
+- an [`renv.lock` file](../../contributing-to-analyses/determining-requirements/determining-software-requirements.md#determining-and-managing-software-dependencies-in-r) if you are working with R, or
+- a [conda `environment.yml`](../../contributing-to-analyses/determining-requirements/determining-software-requirements.md#module-specific-conda-environments) and/or [`conda-lock.yml` file](../../contributing-to-analyses/determining-requirements/determining-software-requirements.md#freezing-dependencies-with-conda-lock) if you are working with Python or other software available through [Bioconda](https://bioconda.github.io).
+
+
+
+You will also need to have [Docker installed on your computer](../../technical-setup/environment-setup/install-docker.md) to build and test the images.
+
+## Docker template files
+
+When you [create an analysis module](../../contributing-to-analyses/analysis-modules/creating-a-module.md), a template `Dockerfile` and a `.dockerignore` file will be created in the root of the analysis module.
+
+The `Dockerfile` includes a `FROM` statement that specifies the base image to use and a couple of `LABEL` statements to populate metadata about the image.
+
+The `.dockerignore` file is used to specify files and directories that should *not* be included in the Docker build environment.
+By default, this will be set to ignore all files except environment files (e.g., `renv.lock`, `environment.yml`, `conda-lock.yml`).
+This will keep the build environment small and prevent unnecessary files from being included in the image.
+
+## Base images
+
+The base image you choose will depend on the software requirements of your analysis module.
+You should aim to use a publicly available base image that is in active development and well-maintained in order to ensure that your image is secure and up-to-date.
+You should always use a *versioned* base image to ensure that your analysis module is reproducible, i.e., not the one tagged as `latest`.
```

Comment:
Maybe just add a note here that we provide recommendations on base images to use for R and python below?